src/matches.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+export type NormalizerFn = (textToNormalize: string) => string;
+
+function matches(
+  textToMatch: string,
+  matcher: string | RegExp,
+  normalizer: NormalizerFn,
+  exact?: boolean = true
+) {
+  if (typeof textToMatch !== 'string') {
+    return false;
+  }
+
+  const normalizedText = normalizer(textToMatch);
+  if (typeof matcher === 'string') {
+    return exact
+      ? normalizedText === matcher
+      : normalizedText.toLowerCase().includes(matcher.toLowerCase());
+  } else {
+    return matcher.test(normalizedText);
+  }
+}
+
+type NormalizerConfig = {
+  trim?: boolean,
+  collapseWhitespace?: boolean,
+};
+
+function getDefaultNormalizer({
+  trim = true,
+  collapseWhitespace = true,
+}: NormalizerConfig = {}): NormalizerFn {
```

Comment:
**thought** if `collapseWhitespace` is true then the value of `trim` can be ignored. Maybe the logic should more clearly reflect this:

```javascript 
if (collapseWhitespace) return text.replace(/\s+/g, ' ')
if (trim) return text.trim()

return text
```

Reply:
If I understand correctly collapeseWhitespace does not imply trim. Am I missing something?

E.g. for input: `"     a    a     "`
- collapseWhitespace: `" a a "`
- trim only: `"a    a"`
- collapseWhitespace + trim: `"a a"`